### PR TITLE
TJ Dashboard framework backlog fixes

### PR DESCRIPTION
**Pull request by monikaatvowel**

### CODE REVIEW

File: packages/tjdashboard/administrator/models/forms/widget.xml [hunk 1]
```diff
@@ -53,10 +98,10 @@
 			class="inputbox"
 			required="true">
 				<option value=""></option>
-				<option value="3">Screen 1</option>
-				<option value="6">Screen 2</option>
-				<option value="9">Screen 3</option>
-				<option value="12">Screen 4</option>
+				<option value="12">Full width</option>
```
Comment (thite-amol) [inline packages/tjdashboard/administrator/models/forms/widget.xml]: Add language constant

File: packages/tjdashboard/administrator/models/forms/widget.xml [hunk 2]
```diff
@@ -8,25 +8,57 @@
...
 		<field
-			name="data_plugin" 
+			name="primary_text"
+			filter="string"
+			type="text"
+			label="COM_TJDASHBOARD_WIDGET_FORM_LBL_PRIMARY_TEXT"
+			description="COM_TJDASHBOARD_WIDGET_FORM_DESC_PRIMARY_TEXT"
+			size="15"/>
+
+		<field
```
Comment (thite-amol) [inline packages/tjdashboard/administrator/models/forms/widget.xml]: @monikaatvowel This should not the property of the widget. The UI is the property of renderer.

cc
@coolbung

File: packages/tjdashboard/administrator/sql/uninstall.mysql.utf8.sql [hunk 3]
```diff
@@ -0,0 +1,2 @@
+DROP IF EXISTS `#__tj_dashboards`;
```
Comment (thite-amol) [inline packages/tjdashboard/administrator/sql/uninstall.mysql.utf8.sql]: @monikaatvowel Check this with vaishali

cc
@twsvaishali

File: packages/tjdashboard/administrator/views/dashboard/tmpl/default.php [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+$document->addStylesheet(Juri::root().'components/com_tjdashboard/assets/css/dashboard.css');
+$document->addStylesheet(Juri::root().'plugins/tjdashboardrenderer/countbox/assets/css/countbox.css');
+$document->addStylesheet(Juri::root().'media/techjoomla_strapper/css/bootstrap.j3.min.css');
+
+$document->addScript(Juri::root().'plugins/tjdashboardrenderer/tabulator/assets/js/renderer.js');
+$document->addScript(Juri::root().'components/com_tjdashboard/assets/js/tjDashboardService.js');
+$document->addScript(Juri::root().'components/com_tjdashboard/assets/js/tjDashboardUI.js');
+
+$document->addScript('https://cdnjs.cloudflare.com/ajax/libs/Chart.js/2.5.0/Chart.min.js');
+$document->addScript(Juri::root().'plugins/tjdashboardrenderer/chartjs/assets/js/renderer.js');
```
Comment (thite-amol) [inline packages/tjdashboard/administrator/views/dashboard/tmpl/default.php]: @monikaatvowel Check the requirement of this view.

File: packages/tjdashboard/administrator/views/dashboard/view.html.php [hunk 5]
```diff
@@ -57,6 +57,12 @@ class TjdashboardViewDashboard extends JViewLegacy
 	 */
 	public function display($tpl = null)
 	{
+		require_once JPATH_COMPONENT . '/../com_tjlms/helpers/tjlms.php';
```
Comment (thite-amol) [inline packages/tjdashboard/administrator/views/dashboard/view.html.php]: @monikaatvowel Remove this from tjdashboard

File: packages/tjdashboard/administrator/views/widgets/tmpl/default.php [hunk 6]
```diff
@@ -7,8 +7,9 @@
  */
 // No direct access
 defined('_JEXEC') or die('Restricted access');
-
-
+$document = JFactory::getDocument();
+$document->addScript('components/com_tjdashboard/assets/js/tjDashboardService.js');
```
Comment (thite-amol) [inline packages/tjdashboard/administrator/views/widgets/tmpl/default.php]: @monikaatvowel Check the use of this JS files

File: packages/tjdashboard/site/assets/js/tjDashboardUI.js [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+ * @version    SVN:<SVN_ID>
+ * @package    com_tjlms
+ * @author     Techjoomla <extensions@techjoomla.com>
+ * @copyright  Copyright (c) 2009-2015 TechJoomla. All rights reserved
+ * @license    GNU General Public License version 2, or later
+ */
+
+var TJDashboardUI = {
+
+	initDashboard : function(id){
```
Comment (thite-amol) [inline packages/tjdashboard/site/assets/js/tjDashboardUI.js]: @monikaatvowel Check the syntax on JSlint. Also create min file.

File: packages/tjdashboard/site/assets/js/tjDashboardUI.js [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+
+	/** global: TJDashboardService */
+	var promise = TJDashboardService.getDashboard(id);
+
+	promise.done(function(response) {
+			if(!response.data.dashboard_id)
+			{
+				alert("no length");
```
Comment (thite-amol) [inline packages/tjdashboard/site/assets/js/tjDashboardUI.js]: @monikaatvowel Remove this

File: packages/tjdashboard/site/assets/js/tjDashboardUI.js [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+				return false;
+			}
+			// console.log(response.data);
+
+
+			jQuery('<h1><div data-dashboard-id="'+response.data.dashboard_id+'" class="tjdashboard-title">' + response.data.title + '</div></h1>').appendTo('.tjdashboard');
+
+			if (response.data.widget_data.length <= 0)
+			{
+				jQuery('<div class="alert alert-info"> No widgets found to show</div>').appendTo('.tjdashboard');
```
Comment (thite-amol) [inline packages/tjdashboard/site/assets/js/tjDashboardUI.js]: @monikaatvowel Add language constant

File: packages/tjdashboard/site/assets/js/tjDashboardUI.js [hunk 10]
```diff
@@ -0,0 +1,201 @@
...
+				return false;
+			}
+
+			var divSpan = 0;
+			var i = 0;
```
Comment (thite-amol) [inline packages/tjdashboard/site/assets/js/tjDashboardUI.js]: @monikaatvowel Use meaningful variables

File: packages/tjdashboard/site/models/dashboard.php [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+ * @author     Techjoomla <extensions@techjoomla.com>
+ * @copyright  2017 Techjoomla
+ * @license    GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+// No direct access.
+
+defined('_JEXEC') or die;
+
+use Joomla\Registry\Registry;
```
Comment (thite-amol) [inline packages/tjdashboard/site/models/dashboard.php]: @monikaatvowel remove unused imports

File: packages/tjdashboard/site/views/dashboard/tmpl/default.php [hunk 12]
```diff
@@ -21,18 +21,29 @@
...
+$document->addStylesheet('components/com_tjdashboard/assets/css/dashboard.css');
+$document->addStylesheet('plugins/tjdashboardrenderer/countbox/assets/css/countbox.css');
+$document->addStylesheet('media/techjoomla_strapper/css/bootstrap.j3.min.css');
+
+
 $document->addScript('plugins/tjdashboardrenderer/tabulator/assets/js/renderer.js');
 $document->addScript('components/com_tjdashboard/assets/js/tjDashboardService.js');
 $document->addScript('components/com_tjdashboard/assets/js/tjDashboardUI.js');
 
+$document->addScript('https://cdnjs.cloudflare.com/ajax/libs/Chart.js/2.5.0/Chart.min.js');
```
Comment (thite-amol) [inline packages/tjdashboard/site/views/dashboard/tmpl/default.php]: @monikaatvowel This should be loaded by renderer.

File: packages/tjdashboard/site/views/dashboard/tmpl/default.xml [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+            name="basic"
+            label="COM_TJDASHBOARD_FIELDSET_ITEM_ID_SELECT_LABEL">
+                <fieldset
+                name="basic"
+                label="COM_TJDASHBOARD_FIELDSET_ITEM_ID_SELECT_LABEL">
+                    <field name="dashboard_id"
+                type="sql"
+                label="COM_TJDASHBOARD_WIDGET_FORM_LBL_DASHBOARDID"
+                description="COM_TJDASHBOARD_WIDGET_FORM_DESC_DASHBOARDID"
+                class="inputbox"
```
Comment (thite-amol) [inline packages/tjdashboard/site/views/dashboard/tmpl/default.xml]: @monikaatvowel Update the xml file as per discussion

File: pkg_tjdashboard.xml [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+	<extension type="package" version="3.0" method="upgrade">
+	<name>TJ Dashboard Package</name>
+	<author>Techjoomla</author>
+	<authorUrl>www.techjoomla.com</authorUrl>
+	<copyright>Copyright(C)2012-17 TechJoomla</copyright>
+	<license>http://www.gnu.org/licenses/gpl-2.0.html GNU/GPL</license>
+	<creationDate>5th December 2017</creationDate>
+	<packagename>TJDashboard</packagename>
+	<version>0</version>
+	<packager>Techjoomla Team</packager>
```
Comment (thite-amol) [inline pkg_tjdashboard.xml]: @monikaatvowel Check the requirement of this file

File: script.pkg_tjdashboard.php [hunk 15]
```diff
@@ -0,0 +1,464 @@
...
+																	'tabulator' => 1,
+																),
+															"api" => array(
+																	"tjdashboard" => 1
+																)
+														)
+									);
+
+	private $uninstall_queue = array(
+										'plugins' => array(
```
Comment (thite-amol) [inline script.pkg_tjdashboard.php]: @monikaatvowel Is this required? If yes then please optimize the code and formatting.

cc
@coolbung @twsvaishali

File: packages/tjdashboard/administrator/models/forms/widget.xml (hunk 2 above)
Comment (monikaatvowel) [reply packages/tjdashboard/administrator/models/forms/widget.xml]: @thite-amol fixed now. the code is moved to renderer side and now using params field

File: packages/tjdashboard/administrator/models/forms/widget.xml (hunk 1 above)
Comment (monikaatvowel) [reply packages/tjdashboard/administrator/models/forms/widget.xml]: done

File: packages/tjdashboard/administrator/sql/uninstall.mysql.utf8.sql (hunk 3 above)
Comment (monikaatvowel) [reply packages/tjdashboard/administrator/sql/uninstall.mysql.utf8.sql]: removed the fields now using params

File: packages/tjdashboard/administrator/views/dashboard/tmpl/default.php (hunk 4 above)
Comment (monikaatvowel) [reply packages/tjdashboard/administrator/views/dashboard/tmpl/default.php]: removed for now

File: packages/tjdashboard/administrator/views/dashboard/view.html.php (hunk 5 above)
Comment (monikaatvowel) [reply packages/tjdashboard/administrator/views/dashboard/view.html.php]: now removed

File: packages/tjdashboard/site/views/dashboard/tmpl/default.php (hunk 12 above)
Comment (monikaatvowel) [reply packages/tjdashboard/site/views/dashboard/tmpl/default.php]: renderer js dynamically loaded now

**Pull request merged**